nevergrad/optimization/base.py

```diff
@@ -503,6 +507,12 @@ def ask(self) -> p.Parameter:
         # make sure to call value getter which may update the value, before we freeze the paremeter
         candidate.value  # pylint: disable=pointless-statement
         candidate.freeze()  # make sure it is not modified somewhere
+        self._internal_timer += time.time() - this_ask_start_time
+        time_multiplier = (self.budget / self._num_ask) if self.budget is not None else 1.0
+        if (
+            time_multiplier * self._internal_timer > 5 * 3600
+        ):  # The internal cost should never exceed 5h, or we switch to economy mode.
+            self._is_bored = True
```

Comment:
this is too specific and should be included in the metamodel itself
also, if we check the duration of our optimizers, we should include the tell as well for consistency (some optimizers could spend more time during tell)